code/game/objects/structures/nest.dm

```diff
@@ -0,0 +1,65 @@
+// Nests spawn mobs and objects when crossed by players.
+// Nests placed in "spawn_trigger_distance" will always spawn the same type of mob.
+
+/obj/structure/nest
+	name = "nest"
+	desc = "A twisted, dark passage to the underground."
+	icon = 'icons/mob/nest.dmi'
+	icon_state = "hole"
+
+	move_resist = INFINITY
+	anchored = TRUE
+	density = FALSE
+
+	var/faction = list("hostile")
+	var/spawn_byproduct = list(/obj/item/stack/ore/glass, /obj/item/stack/ore/iron)
+	var/spawn_byproduct_max = 3
+	var/spawn_max = 2
+	var/spawn_mob_options = list(/mob/living/simple_animal/crab)
+	var/spawn_trigger_distance = 7
+
+/obj/structure/nest/Initialize(mapload)
+	..()
+	RegisterSignal(src, COMSIG_MOVABLE_CROSSED, .proc/try_spawn)
+	desc = "[initial(desc)] You can hear a cacophony of growling snores from within."
```

Comment:
```suggestion
/obj/structure/nest/Initialize(mapload)
	. = ..()
	RegisterSignal(src, COMSIG_MOVABLE_CROSSED, .proc/try_spawn)
	desc = "[desc] You can hear a cacophony of growling snores from within."
```

Is it worth overriding `examine()` to only add this line if it hasnt been spawned?

Reply:
I intend it to be a warning for the players that it has mobs inside it. If this is too taxing or if this is not the right way, I can remove it-

Reply:
Addressed this by overriding `examine()` instead of `desc` in 587c6161feb3fe89cbc5d2686ffebe15c72966df